### PR TITLE
Minor code fixes and blacklist ids field in settings

### PR DESCRIPTION
**Pull request by mvkv**

### CODE REVIEW

File: Instastories.py [hunk 1]
```diff
@@ -76,6 +76,10 @@ def get_ids(stories_ids, user_limit, extra_ids, ids_mode):
     """
     ids = (stories_ids[:user_limit] if ids_mode != "extra_ids_only" else []) + \
           (extra_ids if ids_mode != "stories_ids_only" else [])
+
+    for id in blacklisted_ids:
+        if id in ids:
+            ids.remove(int(id))
```
Comment (powahftw) [inline Instastories.py]: Use list comprehension

File: Instastories.py [hunk 2]
```diff
@@ -61,11 +61,11 @@ def normalize_extra_ids(ids):
             if id_of_nick:
                 cached_ids_to_nick[id_of_nick] = nick
                 nicks_to_ids[nick] = id_of_nick
-        if nick in nicks_to_ids: converted_nicknames.append(nicks_to_ids[nick])
+        if nick in nicks_to_ids: converted_nicknames.append(int(nicks_to_ids[nick]))
     save_cached_ids_to_nick(cached_ids_to_nick)
     return numeric_ids + converted_nicknames
 
-def get_ids(stories_ids, user_limit, extra_ids, ids_mode):
+def get_ids(stories_ids, user_limit, extra_ids, ids_mode, blacklisted_ids):
```
Comment (powahftw) [inline Instastories.py]: Move the params order to put extra_ids and blacklisted_ids close together. (As last 2 param)

File: Instastories.py [hunk 3]
```diff
@@ -290,14 +294,11 @@ def nick_to_id(nickname):
     Args:
         usr_list (string): Username.
     Returns:
-        ids (number): Id
+        ids (number): Id if nickname is valid, else return none
```
Comment (powahftw) [inline Instastories.py]: None, also add before that it can be number | None

File: Instastories.py [hunk 4]
```diff
@@ -319,15 +321,20 @@ def start_scrape(scrape_settings, user_limit, media_mode="all", ids_source="all"
         stories_ids = tray_to_ids(stories)
         if user_limit <= 0: user_limit = len(stories_ids)
     extra_ids = normalize_extra_ids(scrape_settings["extra_ids"] if "extra_ids" in scrape_settings else [])
-    ids = get_ids(stories_ids, user_limit, extra_ids, ids_source)
+    blacklisted_ids = []
```
Comment (powahftw) [inline Instastories.py]: Why not reusing normalize_extra_ids ?

File: static/js/settings.js [hunk 5]
```diff
@@ -130,7 +168,7 @@ const updateStatusBar = (res) => {
     statusBar.innerText = '';
   };
 
-  setTimeout(() => hideStatusBar(statusBar), 5000);
+  setTimeout(() => hideStatusBar(statusBar), 2000);
```
Comment (powahftw) [inline static/js/settings.js]: Create a const instead of hardcoding the value

File: static/js/settings.js [hunk 6]
```diff
@@ -53,6 +55,38 @@ const renderExtraIDs = () => {
   root.appendChild(extraIDsNode);
 };
 
+const addBlacklistedID = (blacklistedID) => {
```
Comment (powahftw) [inline static/js/settings.js]: Reuse the code for handling the extraId, maybe with an extra param. So you can prevent code duplication.

**Review by powahftw — CHANGES_REQUESTED** [review]

File: Instastories.py (hunk 2 above)
Comment (mvkv) [reply Instastories.py]: Done

File: Instastories.py (hunk 1 above)
Comment (mvkv) [reply Instastories.py]: Done

File: Instastories.py (hunk 3 above)
Comment (mvkv) [reply Instastories.py]: Done

File: static/js/settings.js (hunk 5 above)
Comment (mvkv) [reply static/js/settings.js]: Done

File: static/js/settings.js (hunk 6 above)
Comment (mvkv) [reply static/js/settings.js]: Done

File: Instastories.py (hunk 4 above)
Comment (mvkv) [reply Instastories.py]: Done

File: Instastories.py [hunk 7]
```diff
@@ -319,15 +319,16 @@ def start_scrape(scrape_settings, user_limit, media_mode="all", ids_source="all"
         stories_ids = tray_to_ids(stories)
         if user_limit <= 0: user_limit = len(stories_ids)
     extra_ids = normalize_extra_ids(scrape_settings["extra_ids"] if "extra_ids" in scrape_settings else [])
-    ids = get_ids(stories_ids, user_limit, extra_ids, ids_source)
+    blacklisted_ids = normalize_extra_ids(scrape_settings['blacklisted_ids'])
```
Comment (powahftw) [inline Instastories.py]: Maybe rename the function to something both applicable to extra id and blaclisted id? 

like: `normalize_ids`

File: static/js/settings.js [hunk 8]
```diff
@@ -21,20 +23,32 @@ const displayWarningUnsavedChanges = (isActive) => {
   }
 };
 
-const deleteExtraID = (extraID) => {
-  if (extraIDs.delete(extraID)) {
+const deleteID = (id, listType) => {
+  if (listType === 'extraIDs') {
```
Comment (powahftw) [inline static/js/settings.js]: Use ternary operator.

File: static/js/settings.js [hunk 9]
```diff
@@ -21,20 +23,32 @@ const displayWarningUnsavedChanges = (isActive) => {
...
 
-const addExtraID = (extraID) => {
-  if ((extraID.trim().length > 0) && !extraIDs.has(extraID)) {
-    extraIDs.add(extraID);
+const addID = (id, listType) => {
+  if (listType === 'extraIDs') {
+    var IDs = extraIDs;
+  } else if (listType === 'blacklistedIDs') {
+    var IDs = blacklistedIDs;
+  }
```
Comment (powahftw) [inline static/js/settings.js]: Use ternary operator

**Review by powahftw — APPROVED** [review]

File: Instastories.py (hunk 7 above)
Comment (mvkv) [reply Instastories.py]: Done

File: static/js/settings.js (hunk 8 above)
Comment (mvkv) [reply static/js/settings.js]: Done

File: static/js/settings.js (hunk 9 above)
Comment (mvkv) [reply static/js/settings.js]: Done

**Pull request merged**